From ac24d24139c1a2a926fc1934d0b7eed12e69b603 Mon Sep 17 00:00:00 2001
From: Simon Horman <horms+renesas@verge.net.au>
Date: Wed, 3 Jul 2013 13:38:30 +0900
Subject: [PATCH 0969/1083] Revert "drm/rcar-du: Use drm_encoder_cleanup()
 directly as .destroy handlers"

This reverts commit 9fb57b40b125054876c377283a0d66174c11b39e.
(cherry picked from commit 0544e564084cb2279a1cf9b13f388c40a3d1523c)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_lvds.c | 7 ++++++-
 drivers/gpu/drm/rcar-du/rcar_du_vga.c  | 7 ++++++-
 2 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_lvds.c b/drivers/gpu/drm/rcar-du/rcar_du_lvds.c
index efd7e86..841db01 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_lvds.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_lvds.c
@@ -212,8 +212,13 @@ static const struct drm_encoder_helper_funcs encoder_helper_funcs = {
 	.mode_set = rcar_du_lvds_encoder_mode_set,
 };
 
+static void rcar_du_lvds_encoder_destroy(struct drm_encoder *encoder)
+{
+	drm_encoder_cleanup(encoder);
+}
+
 static const struct drm_encoder_funcs encoder_funcs = {
-	.destroy = drm_encoder_cleanup,
+	.destroy = rcar_du_lvds_encoder_destroy,
 };
 
 int rcar_du_lvds_init(struct rcar_du_device *rcdu,
diff --git a/drivers/gpu/drm/rcar-du/rcar_du_vga.c b/drivers/gpu/drm/rcar-du/rcar_du_vga.c
index ac90da8..8b90c71 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_vga.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_vga.c
@@ -158,8 +158,13 @@ static const struct drm_encoder_helper_funcs encoder_helper_funcs = {
 	.mode_set = rcar_du_vga_encoder_mode_set,
 };
 
+static void rcar_du_vga_encoder_destroy(struct drm_encoder *encoder)
+{
+	drm_encoder_cleanup(encoder);
+}
+
 static const struct drm_encoder_funcs encoder_funcs = {
-	.destroy = drm_encoder_cleanup,
+	.destroy = rcar_du_vga_encoder_destroy,
 };
 
 int rcar_du_vga_init(struct rcar_du_device *rcdu,
-- 
1.8.3.2

